app/components/Search/mazemapAutocomplete.js

```diff
@@ -0,0 +1,113 @@
+// @flow
+
+import React, { Component, type ComponentType } from 'react';
+import { debounce } from 'lodash';
+import { stripHtmlTags } from './utils.js';
+import "node_modules/mazemap/mazemap.min.css";
+import * as Mazemap from 'mazemap';
+
+type InjectedProps = {
+  mazemapSearch: (query: string) => Promise<*>,
+  meta: any,
+};
+
+type State = {
+  searching: boolean,
+  result: Array<Object>,
+};
+
+function mazemapAutocomplete<Props>({
+  WrappedComponent,
+}: {
+  WrappedComponent: ComponentType<Props>,
+}) {
+  const displayName =
+    WrappedComponent.displayName || WrappedComponent.name || 'Unknown';
+
+  const mySearch = new Mazemap.Search.SearchController({
+    campusid: 1,
+    rows: 10,
+    withpois: true,
+    withbuilding: false,
+    withtype: false,
+    withcampus: false,
+    resultsFormat: 'geojson',
+  });
+
+  return class extends Component<InjectedProps & Props, State> {
+    static displayName = `Autocomplete(${displayName})`;
+
+    state = {
+      searching: false,
+      result: [], //first result fetched on mount
+    };
+
+    _isMounted = false;
+
+    componentDidMount() {
+      this._isMounted = true;
+      if (this.props.meta.initial) {
+        Mazemap.Data.getPoi(this.props.meta.initial.value).then((poi) =>
+          this.setState({
+            result: [
+              {
+                label:
+                  poi.properties.title + ', ' + poi.properties.buildingName,
+                value: this.props.meta.initial.value,
+              },
+            ],
+          })
+        );
+      }
+    }
+
+    componentWillUnmount() {
+      this._isMounted = false;
+    }
+
+    handleSearch = (query: string): void => {
+      if (!query) {
+        return;
+      }
+      this.setState({
+        searching: true,
+      });
+
+      mySearch
+        .search(query)
+        .then((results) => {
+          if (this._isMounted) {
+            this.setState({
+              result: results.results.features.map((result) => ({
+                label: stripHtmlTags(
+                  result.properties.dispPoiNames[0] +
+                    ', ' +
+                    result.properties.dispBldNames[0]
+                ),
+                value: result.properties.poiId,
+              })),
+              searching: false,
+            });
+          }
+        })
+        .catch(() => {
+          if (this._isMounted) {
+            this.setState({ searching: false });
+          }
+        });
+    };
```

Comment:
I would just both of these in a finally.
```
.finally() => {
          if (this._isMounted) {
            this.setState({ searching: false });
          }
}
```
It makes little difference, but I think it's more clear that way as it shows the reader that a mounted component must set `searching: false` in all cases.